### PR TITLE
[Naming] Foreach over "data", renames to "datum"

### PR DESCRIPTION
**Pull request by dobryy**
Closes: https://github.com/rectorphp/rector/issues/4216

### CODE REVIEW

File: rules/naming/src/Naming/PropertyNaming.php [hunk 1]
```diff
@@ -44,10 +45,10 @@ final class PropertyNaming
      */
     private $inflector;
 
-    public function __construct(TypeUnwrapper $typeUnwrapper, Inflector $inflector)
+    public function __construct(TypeUnwrapper $typeUnwrapper, RectorNamingInflector $rectorNamingInflector)
     {
         $this->typeUnwrapper = $typeUnwrapper;
-        $this->inflector = $inflector;
+        $this->inflector = $rectorNamingInflector->build();
```
Comment (dobryy) [inline rules/naming/src/Naming/PropertyNaming.php:51]: @TomasVotruba, I didn't find (quickly) any better and at the same time easy way to decouple the logic for the creation of customized Inflector. If it looks way too bad from your point of you, I can invest more time into it.

File: rules/naming/src/RectorNamingInflector.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+final class RectorNamingInflector
+{
+    public function build(): Inflector
+    {
+        // @see https://www.doctrine-project.org/projects/doctrine-inflector/en/2.0/index.html
+        return InflectorFactory::createForLanguage(Language::ENGLISH)
+            ->withSingularRules(
+                new Ruleset(
+                    new Transformations(new Transformation(new Pattern('/^packagesData$/i'), 'packageData')),
```
Comment (dobryy) [inline rules/naming/src/RectorNamingInflector.php:25]: BTW, these exceptions could be a rule setting in the future :)

Comment (dobryy) [reply rules/naming/src/RectorNamingInflector.php:25]: or... Set setting.

Comment (TomasVotruba) [reply rules/naming/src/RectorNamingInflector.php:25]: Could you add test for this class? Just to confirm its doing the job right

Comment (TomasVotruba) [reply rules/naming/src/RectorNamingInflector.php:25]: Could you cover " anythingData" in tests too?

Comment (dobryy) [reply rules/naming/src/RectorNamingInflector.php]: In order to make "anythingData" work, I had to change the logic.

**Comment by dobryy** [comment]
@TomasVotruba, ready for review.

**Comment by dobryy** [comment]
BTW, I guess that PHPStan issues are caused by some package update and I don't feel like a boy scout when it comes to 300+ errors :)
I also tried to lock versions of some composer packages but it didn't help.

**Comment by TomasVotruba** [comment]
> BTW, I guess that PHPStan issues are caused by some package update and I don't feel like a boy scout when it comes to 300+ errors :)

I agree. It's more like "a superman scout" ;)

It's usually new rules I add to Symplify set. Just no to forget to apply them here too.
It would be nice to know what rule caused the error in the error formatter, so you know you can skip "Symplify\*" rules.

**Comment by TomasVotruba** [comment]
The feature itself looks good, thanks for more detailed test  :+1:

**Comment by dobryy** [comment]
> > BTW, I guess that PHPStan issues are caused by some package update and I don't feel like a boy scout when it comes to 300+ errors :)
> 
> I agree. It's more like "a superman scout" ;)
> 
> It's usually new rules I add to Symplify set. Just no to forget to apply them here too.
> It would be nice to know what rule caused the error in the error formatter, so you know you can skip "Symplify*" rules.

I noticed the error message of the new rule, is it a common approach that you use in projects? I mean do use factories in all cases instead of plain object creation?

**Comment by TomasVotruba** [comment]
This project is special, so all `*/Node/*` classes should be excluded.
But otherwise the repeated object creation should be decoupled to services. 

Same way logic should not be in Controller/Command/EventDispatcher, but it own service that are only called there.

**Pull request merged**